assignment.md

```diff
@@ -11,63 +11,113 @@ If the version control concepts or `git` commands are still confusing to you, co
-       > Answer here
+       > I see the stashed changes as work in progress; that's the top line of the output.
      - c. Look up what `git stash drop` and `git stash pop` does, either Google or use `--help`. Run `git stash drop` to undo what you did with `stash`, in what scenario might `stash`-ing, `pop`-ing, and `drop`-ing be useful?
-       > Answer here
+       > Stashing is useful when I'm not yet ready to commit my changes (I'm in a draft mode and have not finished but need to work on another file in the same repo before I'm done making changes). Dropping is useful when I'm done working with a stash and don't want to commit. Popping is useful when I'm done working with a stash and do want to commit.
 
  6. One common mistake when learning `git` is to commit large files that should not be managed by `git` or adding sensitive information like security keypairs for Amazon Cloud Services. Navigate back to your assignment repository and try adding a random text file to the repository (use `touch`), making some modifications and commits to your repository, and deleting that file with `rm`. But deleting with `rm` will not delete that file's recorded git history, therefore we need to do a little more. You may want to look at [this](https://help.github.com/articles/removing-sensitive-data-from-a-repository/))
 
-    > Perform the commits and deletion, this question will be graded based on your commit history on GitHub. I should see commits for when you added your file and made modifications, and I should see a commit for when you deleted the files. 
+    > I accidentally did this on main the first time around, but we fixed it together later.
 
 
  7. Like many command line tools, `git` provides a configuration file (or dotfile) called `~/.gitconfig`. Create an alias in `~/.gitconfig` so that when you run `git graph`, you get the output of `git log --all --graph --decorate --oneline`. You can do this by directly modifying the `~/.gitconfig` file or by using the `git` command line functionality.
     ```bash
-    # Your alias entry or git config command here
+    [alias]
+        graph = log --all --graph --decorate --oneline
     ```
     This is also a good opportunity to turn your dotfiles folder into a version controlled repo (Use `git init` in your dotfiles folder), make it private if you have sensitive `ssh` configurations there (IP Addresses), else feel free to make it public. (This is for your own benefit, not part of the assignment grading)
 
  8. You can define global ignore patterns in `~/.gitignore_global` after running `git config --global core.excludesfile ~/.gitignore_global`. Do this, and set up your global `gitignore` file to ignore OS-specific or editor-specific temporary files, like `.DS_Store`, or `.vscode`. Feel free to Google around for common gitignore entries for Python projects or other projects of your interest.
 
     ```bash
-    # Your .gitignore_global entries here
+    .DS_Store
+    .DS_Store?
+    Thumbs.db
```

Comment:
For python projects, it's useful to add `__pycache__`, `.ipynb_checkpoints`, and `.vscode`.

Reply:
Thanks!!!